### PR TITLE
Update CLAS Type

### PR DESCRIPTION
**Pull request by wurzka**
1. Generated new JSON schema automatically
2. Serialised interface with CLASS-type
3. Serialised example class

### CODE REVIEW

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      "! Parameter descriptions
+      parameters  TYPE ty_component_descriptions,
+    END OF ty_event,
+    "! <p class="shorttext">Event Descriptions</p>
+    "! Event descriptions
+    ty_events TYPE SORTED TABLE OF ty_event WITH UNIQUE KEY name.
+
+  TYPES:
+    "! <p class="shorttext">Interface Properties</p>
+    "! Interface properties
```
Comment (schneidermic0) [inline file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap:138]:
```suggestion
    "! <p class="shorttext">Class Properties</p>
    "! Class properties
```

File: file-formats/clas/clas.json [hunk 2]
```diff
@@ -1,78 +1,80 @@
 {
   "$schema": "https://json-schema.org/draft/2020-12/schema",
-  "$id": "http://github.com/SAP/abap-file-formats/file-formats/clas/clas.json",
+  "$id": "https://github.com/SAP/abap-file-formats/blob/main/file-formats/clas/clas.json",
+  "title": "Interface Properties",
+  "description": "Interface properties",
```
Comment (schneidermic0) [inline file-formats/clas/clas.json:5]: Should be `Class Properties`, but it is fixed by the changes to the ABAP Doc in the interface, see also other comment

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      "! Format version
+      "! $required
+      schema               TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! header
+      "! $required
+      header               TYPE zif_aff_types_v1=>ty_header_60_src,
+      "! <p class="shorttext">Class Category</p>
+      "! Class category
+      "! $required
```
Comment (schneidermic0) [inline file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap:150]: In interfaces the category was not required, maybe we should keep this in sync.

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      bsp_application_class          TYPE ty_category VALUE '60',
+      "! <p class="shorttext">Basis class for BSP element handlers</p>
+      "! Basis class for BSP element handlers
+      basis_class_bsp_element_hdlr   TYPE ty_category VALUE '70',
+      "! <p class="shorttext">Web Dynpro runtime object</p>
+      "! Web Dynpro runtime object
+      web_dynpro_runtime_object      TYPE ty_category VALUE '80',
+      "! <p class="shorttext">ESI: Provider interface (generated)</p>
+      "! ESI: Provider interface (generated)
+      esi_provider_intf_generated    TYPE ty_category VALUE '90',
```
Comment (schneidermic0) [inline file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap:59]: Same here. I think this is only relevant for interfaces

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+      "! <p class="shorttext">Area class (shared objects)</p>
+      "! Area class (shared objects)
+      area_class_shared_objects      TYPE ty_category VALUE '45',
+      "! <p class="shorttext">Test class (ABAP Unit)</p>
+      "! Test class (ABAP Unit)
+      testclass_abap_unit            TYPE ty_category VALUE '05',
+      "! <p class="shorttext">Interface for parameter types for database procedure proxies</p>
+      "! Interface for parameter types for database procedure proxies
+      intf_for_parameter_types       TYPE ty_category VALUE '65',
```
Comment (schneidermic0) [inline file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap:68]: Same here. I think this is only relevant for interfaces

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks. :) See my comments

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap (hunk 1 above)
Comment (wurzka) [reply file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap:138]: Done.

**Comment by wurzka** [comment]
I deleted the categories which were not used in ER9.

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap (hunk 3 above)
Comment (wurzka) [reply file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap]: I deleted the $required statement

File: file-formats/clas/clas.json [hunk 6]
```diff
@@ -1,135 +1,121 @@
...
-        "ViewClass",
-        "ProxyClassForRemoteInterface",
-        "RFCProxyClass",
-        "ExceptionClass",
-        "BusinessClass",
-        "BusinessClassInterfaceForStaticComponents",
-        "BusinessClassInterfaceForInstance-dependentComponents",
-        "BSPApplicationClass",
-        "BasisClassForBSPElementHandlers",
-        "WebDynproRuntimeObject",
```
Comment (crasovan) [inline file-formats/clas/clas.json:70]: we mentioned webDynproRuntimeObject category  for interfaces, should it be for classes mentioned too?

**Review by crasovan — CHANGES_REQUESTED** [review]
as said I'm not sure about category 80 for classes

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap [hunk 7]
```diff
@@ -39,33 +33,18 @@ INTERFACE zif_oo_aff_clas_v1 PUBLIC.
...
-      business_class_intf_for_instan TYPE ty_category VALUE '52',
       "! <p class="shorttext">BSP application class</p>
       "! BSP application class
       bsp_application_class          TYPE ty_category VALUE '60',
       "! <p class="shorttext">Basis class for BSP element handlers</p>
       "! Basis class for BSP element handlers
       basis_class_bsp_element_hdlr   TYPE ty_category VALUE '70',
-      "! <p class="shorttext">Web Dynpro runtime object</p>
-      "! Web Dynpro runtime object
-      web_dynpro_runtime_object      TYPE ty_category VALUE '80',
```
Comment (schneidermic0) [inline file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap]: We should add this. This is valid at least for generated web dynpro runtime classes

**Review by schneidermic0 — CHANGES_REQUESTED** [review]

File: file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap (hunk 7 above)
Comment (wurzka) [reply file-formats/clas/type/zif_oo_aff_clas_v1.intf.abap]: Done

**Review by crasovan — APPROVED** [review]
ok

**Review by schneidermic0 — APPROVED** [review]
LGTM
Thanks :)

**Pull request merged**